api/src/pcapi/core/offerers/api.py

```diff
@@ -2487,42 +2519,64 @@ def update_offerer_address_label(offerer_address_id: int, new_label: str) -> Non
         raise exceptions.OffererAddressLabelAlreadyUsed()
 
 
-def get_or_create_address(address_info: AddressInfo) -> geography_models.Address:
-    departmentCode = utils_regions.get_department_code_from_city_code(address_info.citycode)
-    timezone = date_utils.get_department_timezone(departmentCode)
+LocationData = typing.TypedDict(
+    "LocationData",
+    {
+        "street": str | None,
+        "city": str,
+        "postal_code": str,
+        "insee_code": str | None,
+        "latitude": float,
+        "longitude": float,
+        "ban_id": str | None,
+    },
+)
+
+
+def get_or_create_address(location_data: LocationData) -> geography_models.Address:
+    department_code = None
+    timezone = None
+    insee_code = location_data.get("insee_code")
+    latitude = location_data["latitude"]
+    longitude = location_data["longitude"]
+    street = location_data.get("street")
+    ban_id = location_data.get("ban_id")
+    if insee_code:
+        department_code = utils_regions.get_department_code_from_city_code(insee_code)
+        timezone = date_utils.get_department_timezone(department_code)
```

Comment:
Tu as précédemment écrit « city_code can be either the postalCode or the inseeCode » dans la documentation de `get_department_code_from_city_code`. Il serait donc selon moi préférable d'utiliser le code postal lorsque le code Insee de la commune n'est pas connu. Cela permettrait de s'assurer que `timezone` n'est jamais vide.
```suggestion
    department_code = utils_regions.get_department_code_from_city_code(insee_code or location_data["postal_code"])
    timezone = date_utils.get_department_timezone(department_code)
```

Reply:
Alors la timezone ne peut pas être nulle, par contre effectivement la valeur par défaut va être la TZ métropolitaine. J'ai fait les modifications et rajouté un test